components/KiboGallery.vue

```diff
@@ -0,0 +1,318 @@
+<template>
+  <div class="sf-gallery">
+    <div class="sf-gallery__stage">
+      <div ref="glide" class="glide">
+        <div class="glide__track" data-glide-el="track">
+          <ul class="glide__slides">
+            <li
+              v-for="(picture, index) in images"
+              :key="'slide-' + index"
+              class="glide__slide"
+              @mouseover="startZoom(picture)"
+              @mousemove="moveZoom($event, index)"
+              @mouseout="removeZoom(index)"
+            >
+              <SfImage
+                ref="sfGalleryBigImage"
+                class="sf-gallery__big-image"
+                :class="{ 'sf-gallery__big-image--has-zoom': enableZoom }"
+                :src="picture.desktop.url"
+                :alt="picture.alt"
+                :width="imageWidth"
+                :height="imageHeight"
+                @click="$emit('click:stage', { picture, index })"
+              />
+            </li>
+          </ul>
+        </div>
+      </div>
+      <transition name="sf-fade">
+        <div
+          ref="outSide"
+          :class="{
+            'display-none': !outsideZoom || !pictureSelected || (!outsideZoom && !enableZoom),
+          }"
+          :style="{ width: `${imageWidth}px`, height: `${imageHeight}px` }"
+        >
+          <SfImage
+            ref="imgZoom"
+            class="sf-gallery__zoom"
+            :src="pictureSelectedUrl"
+            :width="imageWidth"
+            :height="imageHeight"
+            :lazy="false"
+            :alt="pictureSelected.alt"
+          />
+        </div>
+      </transition>
+    </div>
+    <div class="sf-gallery__thumbs">
+      <slot name="thumbs" v-bind="{ images, active: activeIndex, go }">
+        <SfButton
+          v-for="(image, index) in images"
+          :key="'img-' + index"
+          class="sf-button--pure sf-gallery__item"
+          :class="{ 'sf-gallery__item--selected': index === activeIndex }"
+          :aria-label="'Image ' + index"
+          @click="go(index)"
+        >
+          <SfImage
+            class="sf-gallery__thumb"
+            :src="image.mobile.url"
+            :alt="image.alt"
+            :width="thumbWidth"
+            :height="thumbHeight"
+          />
+        </SfButton>
+      </slot>
+    </div>
+    <div class="sf-gallery__icons">
+      <div
+        v-for="(image, index) in images"
+        :key="'img-' + index"
+        class="icon"
+        :class="{ 'icon--selected': index === activeIndex }"
+        @click="go(index)"
+      ></div>
+    </div>
+  </div>
+</template>
+<script>
+import Glide from "@glidejs/glide"
+import { SfButton, SfImage } from "@storefront-ui/vue"
+import PinchZoom from "pinch-zoom-js"
+
+export default {
```

Comment:
convert component from options to  composition api

Reply:
Actually we had used code from Storefront UI as it it, but not its converted to composition api